src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -3423,13 +3420,14 @@ class C
 
                 state.SendTypeChars("Sys")
                 Await state.WaitForAsynchronousOperationsAsync()
-                Await state.AssertCompletionSession()
+                Await state.AssertSelectedCompletionItem(displayText:="System")
                 state.SendTypeChars(".")
                 Assert.Contains("System.", state.GetLineTextFromCaretPosition())
             End Using
         End Function
 
-        <MemberData(NameOf(AllCompletionImplementations))>
+        <InlineData(CompletionImplementation.Legacy)>
+        <InlineData(CompletionImplementation.Modern, Skip:="https://github.com/dotnet/roslyn/issues/29112")>
```

Comment:
these need to be unskipped.  Note that this affects partner teams (F# in particular).

Reply:
Thank you! The test was added by TypeScript team in August 2018. We're working with them on adjusting expectations for Non-blocking scenarios.

Here is our dashboard to track open issues: [https://github.com/dotnet/roslyn/projects/38](https://github.com/dotnet/roslyn/projects/38)

The skip should not block merging the PR under a feature flag turned off.

---
In reply to: [232508694](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508694) [](ancestors = 232508694)

Reply:
> We're working with them on adjusting expectations for Non-blocking scenarios.

FWIW, i believe this situatoin would be blocking (no pun intended) for F# (not typescirpt).  Have things been communicated with them?  Are they ok with this under a feature-flag? 

Thanks!

Reply:
Under the feature flag, it does not affect current scenarios.
Adding @brettfo to validate if changes with BlockForCompletionItems affect F#

---
In reply to: [232835505](https://github.com/dotnet/roslyn/pull/29016#discussion_r232835505) [](ancestors = 232835505)

Reply:
Note: iirc, there were two reasons F# needed this.  First, the F# compiler itself isn't designed to be able to answer questions as quickly as C#/VB can.  So blocking to wait on the compiler is just bad for normal code.  Second, F# literally has "type providers" where arbitrary code can run (including making arbitrary network calls).  So blocking here can be really bad.

Reply:
Thank you,@cyrusnajmabadi ! We will validate with F# as well. 

The new completion provides non-blocking for all async methods +TryCommit. 

It does not provide non-blocking for Initialize and ShouldCommitCompletion assuming them to be fast. They added telemetry to measure those methods.

However, TestNoBlockOnCompletionItems3 and TestNoBlockOnCompletionItems4 are special cases: they involve SendCommitUniqueCompletionListItem. The current completion tries to calculate a model asynchronously in the case and if nothing happens in between, makes a commit. We consider if this is the right option or if we should block UI for this specific scenario. @amadeusw leads the discussion and can provide more information.

Here is an issue to track: [https://github.com/dotnet/roslyn/issues/30189](https://github.com/dotnet/roslyn/issues/30189)

This issue does not block merging under a feature flag.

---
In reply to: [232894336](https://github.com/dotnet/roslyn/pull/29016#discussion_r232894336) [](ancestors = 232894336)

Reply:
Will re-iterate this under the next loop of partner signoff. [https://github.com/dotnet/roslyn/issues/30183](https://github.com/dotnet/roslyn/issues/30183)

---
In reply to: [232838251](https://github.com/dotnet/roslyn/pull/29016#discussion_r232838251) [](ancestors = 232838251,232835505)